components/leaderboard/CountdownTimer.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react'
+import { CustomBox } from 'components/OffsetBorder'
+import { intervalToDuration, nextMonday, set } from 'date-fns'
+import type { Duration } from 'date-fns'
+
+const customFormatDuration = (x: Duration) =>
+  `${x.days ? x.days + 'd : ' : ''}${x.hours ? x.hours + 'hr : ' : ''}${
+    x.minutes ? x.minutes + 'min : ' : ''
+  }${x.seconds ? x.seconds + 'sec' : '0sec'}`
+
+const CountdownTimer = () => {
+  const now = new Date()
+  const [$time, $setTime] = useState(now)
+  const end = nextMonday(
+    set(now, { hours: 0, minutes: 0, seconds: 0, milliseconds: 0 })
+  )
```

Comment:
It looks like the countdown is displaying the time until midnight PST instead of midnight UTC -- there probably needs to be some UTC conversion done here or somewhere.